src/utils/color.js

```diff
@@ -1,2 +1,24 @@
-export const foregroundColorFromBackground = (color: string) =>
+const foregroundColorFromBackground = (color: string) =>
   (parseInt(color[1], 16) + parseInt(color[3], 16) + parseInt(color[5], 16) > 23 ? 'black' : 'white');
+
+function shadeColor(color, percent) {
```

Comment:
We should just use a library that does that already.
The best I could find is:
https://github.com/Qix-/color